addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -86,6 +86,14 @@ const LinkTools = Link.extend({
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @override
+     */
+    _onAnyChange: function (e) {
```

Comment:
We cannot do that on any changes as it would create a mutation that is "invisible" and make difficult to understand why the "undo/redo" does not "undo/redo" sometimes.

Reply:
Actually, as we discussed in the office, we should handle this fix differently.
The reason the link is not saved is because the method destroy in the `link_tools.js` call `this._super` after removing the link. It should call it before.

The new code in `link_tools.js` should be:
```js
    destroy: function () {
        if (!this.el) {
            return this._super(...arguments);
        }
        this._super(...arguments);
        const $contents = this.$link.contents();
        if (!this.$link.attr('href') && !this.colorCombinationClass) {
            $contents.unwrap();
        }
        this._observer.disconnect();
        this._removeHintClasses();
    },
```